### DISCUSSION TITLE
[QUESTION] I added a new node to the kubernetes cluster, but longhorn doesn't recognize it. How do I make it recognize this new node?

### DISCUSSION THREAD

**Discussion by michaelmoreira** [longhorn/longhorn#4434] [General]
## Question
> Suggest to use https://github.com/longhorn/longhorn/discussions to ask questions.
I added a new node to the kubernetes cluster, but longhorn doesn't recognize it. How do I make it recognize this new node?

## Environment

 - Longhorn version: 1.3.0
 - Kubernetes version:
 - Node config
   - OS type and version SLES15 SP3
   - CPU per node: 64
   - Memory per node: 164
   - Disk type SSD
   - Network bandwidth and latency between the nodes:
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): Rancher

**Comment by derekbit** [comment]
Can you check if the node is already added to the k8s cluster by `kubectl get nodes` and the hindrance caused by node taint? 
Longhorn system should automatically add this node into its cluster.

**Reply by michaelmoreira** [reply]
Hello, when running kubectl get nodes it returns the 3 initial nodes and 2 nodes that were recently added to the cluster, but longhorn only shows the 3 initial nodes, is there any procedure for it to recognize these 2 nodes that were added?

**Reply by derekbit** [reply]
Can you show the node taint of the newly added nodes?

**Reply by tomclark** [reply]
Hi there,

I'm also having this problem. I created a two node cluster with k3s, did a vanilla install of Longhorn with Helm, and having added a third node, it's not being picked up by Longhorn. Googling around, it sounds very like this issue:

https://github.com/longhorn/longhorn/issues/4352

But I don't have any taints (as far as I'm aware; none are listed when I describe the node though I'm not positive that's what you meant).

Three pods have started on the new node: a longhorn-manager pod (stuck in init; log says "wait-longhorn-admission-webhook waiting"), a longhorn-csi-plugin pod (CrashLoopBackOff; log says "Failed to initialize Longhorn API client: Get \"http://longhorn-backend:9500/v1", which I think is it trying to talk to the manager but failing?), and an engine-image-ei pod, which is running okay.

The most promising of these would seem to be the manager - I assume if I can fix that, the CSI plugin will then work. I can see the admission-webhook pods running on the initial two nodes, which I guess is what the manager is waiting on (speculating at this point!).

Any suggestions most welcome. I have volumes attached and it'd be very inconvenient to need to remove them and start again, but that feels like my best option at the moment - when I'm sure there's a way to make this work.

Edit: the version listed in the dashboard is 1.3.0, it's K3s installed as v1.24.4+k3s1, and Ubuntu 20.04.5. It's running on dedicated hosts from Hetzner. No issues at all with the first two boxes - it's been great, in fact.

**Comment by tomclark** [comment]
I did have the node cordoned, possibly when I did the install. Now that I've uncordoned it, is there something I need to restart?

Going back to my thinking on the admission-webhook pod that's not running on the new node, I can see the replicaset is marked as two desired, which I assume should be three. I just tried updating it to be three, but it did seem to do anything.

**Comment by tomclark** [comment]
I found my solution - and it's entirely user error.

I decided to look at the syslog on the box that wasn't being recognised, and noticed a bunch of these:

```
Sep  4 00:51:44 k3s003p k3s[1052]: I0904 00:51:44.164775    1052 trace.go:205] Trace[1006818408]: "Proxy via http_connect protocol over tcp" address:10.42.0.6:4443 (04-Sep-2022 00:49:33.556) (total time: 130608ms):
Sep  4 00:51:44 k3s003p k3s[1052]: Trace[1006818408]: [2m10.608614609s] [2m10.608614609s] END
```

I decided to shell onto a working pod on the new node and ping that IP address (10.42.0.6). It didn't work. I checked that I could ping it from the other two nodes and it did work, so I then decided to double check my firewall settings, and during the Ansible run to set up the rules, it appears I'd somehow failed to allow the address of the new node to the other two nodes on the public address. I added the public address of the new node to the firewall allow list on the existing nodes, and it all started working. My volumes all started replicating and the dashboard is now a sea of green.

In summary, my issue was as simple as incorrect firewall settings. Doh!
